### PR TITLE
MAGECLOUD-3048: Allow setting of x-frame options within configuration…

### PR DESCRIPTION
**Pull request by shiftedreality**
… file

### Fixed Issues (if relevant)

1. https://magento2.atlassian.net/browse/MAGECLOUD-3048

### Manual testing scenarios

1. https://jira.corp.magento.com/browse/MAGETWO-97626

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Config/Schema.php [hunk 1]
```diff
@@ -406,6 +406,15 @@ public function getSchema()
                     StageConfigInterface::STAGE_POST_DEPLOY => [''],
                 ],
             ],
+            StageConfigInterface::VAR_X_FRAME_CONFIGURATION => [
+                self::SCHEMA_TYPE => ['string'],
+                self::SCHEMA_STAGE => [
+                    StageConfigInterface::STAGE_GLOBAL
+                ],
+                self::SCHEMA_DEFAULT_VALUE => [
+                    StageConfigInterface::STAGE_GLOBAL => ''
```
Comment (billygilbert) [inline src/Config/Schema.php]: looks like default should be SAMEORIGIN.  That's what I get in my env.php without changing anything.

**Review by billygilbert — CHANGES_REQUESTED** [review]
Also needs an update to dist/.magento.env.yaml

File: src/Config/Schema.php (hunk 1 above)
Comment (shiftedreality) [reply src/Config/Schema.php]: This will be backward incompatible because will overwrite custom values.

**Comment by YPyltiai** [comment]
@shiftedreality / @billygilbert - would it be better to move it into `system` section? Doesn't logically fit into `global` (no effect on deployment process).  It's not a requirement - just a question.

**Comment by shiftedreality** [comment]
@billygilbert fix both comments, changing this value by default seems not very risky.
@YPyltiai it's on the same level as cache settings for example

**Comment by YPyltiai** [comment]
Then it's fine.

**Comment by billygilbert** [comment]
Maybe at some point it makes sense to move these all to an `SYSTEM>ENV` section...or something that denotes it specifically affects env.php but I agree with @shiftedreality for now.  Approved.

**Review by billygilbert — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**